src/node/tendermint_internals.ml

```diff
@@ -0,0 +1,158 @@
+open Crypto
+open Protocol
+open Validators
+
+type value =
+  | Block of Protocol.Block.t
+  | Nil
+[@@deriving yojson]
+
+let string_of_value = function
+  | Nil -> "nil"
+  | Block b ->
+    Printf.sprintf "block %s" (Crypto.BLAKE2B.to_string b.Protocol.Block.hash)
+
+let repr_of_value v = v
+
+let produce_value : (State.t -> value) ref = ref (fun _ -> assert false)
+
+let is_valid state value =
+  match value with
+  | Nil -> false
+  | Block block ->
+    let all_operations_properly_signed _block = true in
```

Comment:
?

Reply:
My recommendation is a `FIXME:` comment to preempt our internal alarm bells.